runner/src/server/plugins/engine/components/NumberField.ts

```diff
@@ -49,9 +51,12 @@ export class NumberField extends FormComponent {
 
   getViewModel(formData: FormData, errors: FormSubmissionErrors) {
     const schema: any = this.schema;
+    const options: any = this.options;
     const viewModel = {
       ...super.getViewModel(formData, errors),
       type: "number",
+      ...(options.prefix ? { prefix: { text: options.prefix } } : null),
+      ...(options.suffix ? { suffix: { text: options.suffix } } : null),
```

Comment:
whenever there's a ternary that does nothing (e.g. return null) I'd consider using && instead. 

```suggestion
      ...(options.prefix && { prefix: { text: options.prefix } }),
      ...(options.suffix && { suffix: { text: options.suffix } }),
```

also, are you planning for it to do anything other than display text? maybe a simple string would do? 

BY the way, great minds, I actually opened a PR for a similar feature for react-USWDS for exactly suffix/prefix. They wanted to use suffixOrPrefix (so who am I to argue) - but we could use `affix` which encapsulates prefix/suffix. 

```
const viewModel = {
  //...
  prefixText: options.prefix,
  suffixText: options.suffix,
}
```

but.. I don't think you would have both at the same time? so you could do

```
const { affixType, affixText } = options;
const viewModel = {
  //...
  affixType,
  affixText
}
```

anyway, I'll let you decide which _feels_ best. but `{ prefix: { text: options.prefix } }` is too much nesting on one line making it super hard to read.

Reply:
Thanks for the very detailed review 😄 

We may sometimes to use both a prefix and suffix - for instance in the case where we have something like:

Please enter how much you'd like to charge per egg 🥚 

| £ | [            ] | per 🥚 |

I'll use the actual view to pass in the data - rather then getting it into macro ready form at the component level. 

I agree about how much is being squeezed onto a single line...time to make it beautiful 🎨

Reply:
Perhaps I've misunderstood something 😵‍💫 - but does viewModel not need to exactly match the data which is passed into the macro? Thanks in advance 👍 !!

Reply:
Adoy, course! suffix/prefix it is

Reply:
Although I find it interesting that this pattern is allow by the gov uk design system given the prefix and suffix is ignored by screen readers (iirc). 

It seems to be recommended that the suffix and prefix just repeat information from the form question. 

Do you know the rational behind the US gov design systems allowance of only one affix? Seems like an strange limitation.

# Some concerns.

It does raise the question of implementing a money input which has some accessibility features which make the input more screen reader friendly - and designer friendly since they won't have to also set the precision to 2.

Shall I add a warning (or hint text) for this?

Reply:
Oh no, I suggested affix, I think although they liked it, it was too far removed from the vanilla (non react) USWDS

Reply:
Also noted on precision. Let’s open another issue and discuss and not increase the scope of this too much

Reply:
Ill open an issue now :) let me know if you're satisfied with my changes!